tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Redis/ServiceStack/RedisNativeClientSendReceiveIntegration_6_2_0.cs

```diff
@@ -0,0 +1,99 @@
+        IntegrationName = IntegrationName)]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    // ReSharper disable once InconsistentNaming
+    public class RedisNativeClientSendReceiveIntegration_6_2_0
+    {
+        private const string IntegrationName = nameof(Configuration.IntegrationId.ServiceStackRedis);
+        private const IntegrationId IntegrationId = Configuration.IntegrationId.ServiceStackRedis;
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TFunc">Type of the result processor</typeparam>
+        /// <typeparam name="TAction">Type of the server end point</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="cmdWithBinaryArgs">Cmd with binary args</param>
+        /// <param name="fn">Function instance</param>
+        /// <param name="completePipelineFn">Complete pipeline function instance</param>
+        /// <param name="sendWithoutRead">Send without read boolean</param>
+        /// <param name="operation">The calling member name</param>
+        /// <returns>Calltarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget, TFunc, TAction>(TTarget instance, byte[][] cmdWithBinaryArgs, TFunc fn, TAction completePipelineFn, bool sendWithoutRead, string operation)
+            where TTarget : IRedisNativeClient
+        {
+            Scope scope = RedisHelper.CreateScope(Tracer.Instance, IntegrationId, IntegrationName, instance.Host ?? string.Empty, instance.Port.ToString(CultureInfo.InvariantCulture), GetRawCommand(cmdWithBinaryArgs));
+            if (scope is not null)
+            {
+                return new CallTargetState(scope);
+            }
+
+            return CallTargetState.GetDefault();
+        }
+
+        /// <summary>
+        /// OnMethodEnd callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TResponse">Type of the response</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="response">Response instance</param>
+        /// <param name="exception">Exception instance in case the original code threw an exception.</param>
+        /// <param name="state">Calltarget state value</param>
+        /// <returns>A response value, in an async scenario will be T of Task of T</returns>
+        internal static CallTargetReturn<TResponse> OnMethodEnd<TTarget, TResponse>(TTarget instance, TResponse response, Exception exception, in CallTargetState state)
+        {
+            state.Scope.DisposeWithException(exception);
+            return new CallTargetReturn<TResponse>(response);
+        }
+
+        private static string GetRawCommand(byte[][] cmdWithBinaryArgs)
```

Comment:
Nipticking -  you could have moved this code to RedisHelper instead of duplicating it.

Reply:
Done in b146df7c4c5e9ba61d404b473f6f22d43db00a68